mininet/wifiMobility.py

```diff
@@ -180,7 +180,8 @@ def associate_wpa(self, sta, ap, wlan):
         :param wlan: wlan ID
         """
         passwd = self.verifyPasswd(sta, ap, wlan)
-        os.system('pkill -f \'wpa_supplicant -B -Dnl80211 -i %s\'' % sta.params['wlan'][wlan])
+        confname = "mn%d_%s_wifiDirect.conf" % (os.getpid(), sta.name)
```

Comment:
Sorry. Shouldn't that row kill a specific process for a specific station and interface name?